types/index.test-d.ts

```diff
@@ -0,0 +1,33 @@
+import { IncomingMessage, ServerResponse, createServer } from "node:http";
+import { expectType } from "tsd";
```

Comment:
I'd recommend https://www.npmjs.com/package/expect-type instead.
It has zero dependencies and does not install a fork of typescript, @Phillip9587

Reply:
It would be enough to run `tsd --noEmit` to verify all the assetions.